overlord/snapshotstate/snapshotstate.go

```diff
@@ -69,6 +73,70 @@ func allActiveSnapNames(st *state.State) ([]string, error) {
 	return names, nil
 }
 
+func automaticSnapshotExpiration(st *state.State) time.Duration {
+	// TODO: get from config
+	return DefaultAutomaticSnapshotExpiration
+}
+
+// saveExpiration saves expiration date of the given snapshot set in the state.
+// The state needs to be locked by the caller.
+func saveExpiration(st *state.State, setID uint64, expiryTime time.Time) error {
+	// Expirations actually map to time.Time, use string here as we don't need parsed times here.
+	var expirations map[uint64]string
+	err := st.Get("snapshots-expiry", &expirations)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if expirations == nil {
+		expirations = make(map[uint64]string)
+	}
+	expirations[setID] = expiryTime.Format(time.RFC3339)
+	st.Set("snapshots-expiry", expirations)
+	return nil
+}
+
+// removeExpirations removes expirations of the given set IDs from the state.
+func removeExpirations(st *state.State, setIDs ...uint64) error {
+	// Expirations actually map to time.Time, use string here as we don't need parsed times here.
```

Comment:
Same comment as above, if we keep it this way I would expand the comment.

Reply:
Removed, per above comment.